pkg/load/agent_utils.go

```diff
@@ -37,7 +38,12 @@ func reloadWatcher(ctx context.Context, w *fsnotify.Watcher, root string, errFun
 			return
 		case event := <-w.Events:
 			log.Tracef("Received %v event for %s", event.Op, event.Name)
-			go c.Run()
+			go func() {
+				err := c.Run()
+				if err != nil {
+					logrus.WithError(err).Errorf("Coalescer function failed")
```

Comment:
Do we care to track this in metrics?

Reply:
We probably should. That function already has `errFunc`, which is the func to report the error to the resolver metrics, so that should be easy to implement.